### DISCUSSION TITLE
Aggiornamento alla 2021.10 fallito

### DISCUSSION THREAD

**Discussion by riccionese** [indomus/forum#855] [HUB: Home Assistant]
ho HA Core su pi4 alla core-2021.9.7
ho sempre cercato di aggiornare tempestivamente le release rilasciate ma con la 2021.10 non riesco 
ho gli stessi errori che sono stati segnalati anche sul forum di home-assistant (https://github.com/home-assistant/core/issues/57307) che non mi pare abbia ancora avuto risposta.

qualcun altro ha avuto e magari risolto questo problema ?

**Comment by Scialla** [comment]
Che installazione hai fatto?
Hai installato Ha Core in Venv o in docker?
Se ti dà gli errori del topic linkato presumo tu abbia installato HA core in Venv.

Se così é, inizia ad aggiornare pip nel Venv quindi, da una sessione ssh, entra nel Venv dove hai installato HA e digita:
`python -m pip install pip --upgrade` (Da notare che quando poi si installerà HA farà un downgrade di pip)

per installare cryptography devi installare rust, il compilatore; inizia a installare i setup_tools (che potrebbero tornare utili) con:
`python -m pip install setuptools_rust`

installa rush, usa questo comando:
`curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`

Poi devi uscire da quella sessione ssh e rientrare (per caricare il nuovo path), oppure apri una nuova sessione e, sempre da dentro il Venv di HA, fai ripartire l'aggiornamento di HA, oppure installa prima a mano cryptography con il comando:
`python -m pip install cryptography==3.4.8`
(ci metterà un pò a compilare ed installare cryptography)
poi aggiorna HA....

**Reply by riccionese** [reply]
si.
è venv  ... per ora
sto brigando per passare in docker. prima o poi lo finisco anche :-) 
appena riesco provo a seguire le tue indicazioni

ma ... capitato solo a me ? ho sempre seguito gli aggiornamenti e le guide.

**Reply by Scialla** [reply]
no, capitato a quasi tutti, anche a me..

**Reply by riccionese** [reply]
ok. fatto.
sono alla core-2021.10.3

Grazie

**Comment by Danielsound** [comment]
Stesso problema anche per la mia installazione Core in Venv.
Risolto eseguendo i comandi specificati da Scialla.
GRAZIE!

**Comment by sp2003** [comment]
Ciao a tutti. ho anch'io problemi con rust. in sostanza scarica tutti i componenti, ma all'installazione dell'ultimo 
```
info: installing component 'rustc'
 91.6 MiB / 105.2 MiB ( 87 %)   2.6 MiB/s in 43s ETA:  5sKilled

```
ripulita la cartelle download e riprovato più volte con interruzione a varie percentuali dinstallazione, anhe 99%.
ovviamente prima di lanciare il curl 

```
sudo systemctl stop home-assistant@homeassistant
sudo -u homeassistant -H -s
source /srv/homeassistant/bin/activate
```

cosa può essere??

in alternativa si può installare dai sorgenti, essendoci un file install.sh , ma come lo eseguo con privilegi di root da homeassistant?

**Comment by Scialla** [comment]
Strano si interrompa l'installazione. Ovviamente hai aggiornato pip prima e phyton è superiore alla 3.8

Prova a fare anche una installazione custom di rust senza usare quella di default usando arm-unknown-linux-gnueabihf

Potrebbe anche essere un problema di RAM, guarda questa issue:
https://github.com/rust-lang/rustup/issues/2394

**Reply by sp2003** [reply]
si si, avevo visto questa issue, ma non credo sia un mio problema. cmq ho aperto una discussione sull'installazione di rust a parte un po più completa. se ne hai voglia puoi passare a darmi un consiglio

**Comment by nikpait** [comment]
Ciao, anche io ho lo stesso problema.
Ho seguito la guida passo passo, ma quando eseguo il comando 'python -m pip install cryptography==3.4.8' ricevo questo messaggio id errore:

DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
Defaulting to user installation because normal site-packages is not writeable
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
ERROR: Could not find a version that satisfies the requirement cryptography==3.4.8 (from versions: 0.1, 0.2, 0.2.1, 0.2.2, 0.3, 0.4, 0.5, 0.5.1, 0.5.2, 0.5.3, 0.5.4, 0.6, 0.6.1, 0.7, 0.7.1, 0.7.2, 0.8, 0.8.1, 0.8.2, 0.9, 0.9.1, 0.9.2, 0.9.3, 1.0, 1.0.1, 1.0.2, 1.1, 1.1.1, 1.1.2, 1.2, 1.2.1, 1.2.2, 1.2.3, 1.3, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.4, 1.5, 1.5.1, 1.5.2, 1.5.3, 1.6, 1.7, 1.7.1, 1.7.2, 1.8, 1.8.1, 1.8.2, 1.9, 2.0, 2.0.1, 2.0.2, 2.0.3, 2.1, 2.1.1, 2.1.2, 2.1.3, 2.1.4, 2.2, 2.2.1, 2.2.2, 2.3, 2.3.1, 2.4, 2.4.1, 2.4.2, 2.5, 2.6, 2.6.1, 2.7, 2.8, 2.9, 2.9.1, 2.9.2, 3.0, 3.1, 3.1.1, 3.2, 3.2.1, 3.3, 3.3.1, 3.3.2)
ERROR: No matching distribution found for cryptography==3.4.8

La versione di Pyton installata è la 3.8.6

Grazie

**Reply by snakuzzo** [reply]
Eh no...
Domanda...ma il log lo hai almeno letto? :D
Stai eseguendo l'installazione con Python 2.7 e non con la 3.8.6.

Se lo esegui da fuori il venv non avresti comunque il modulo cryptography nel venv di HA
Inoltre, come hai appena mostrato, il symlink `python` punta al binario della versione 2.7. Quello che punta alla 3.x è `python3`.

Il comando di installazione va eseguito attivando il venv in cui hai installato Home Assistant.

Prima di proseguire vi consiglio di chiarirvi alcuni concetti su Python e su come funzionano i Virtual Environment altrimenti si rischia di fare cose a caso senza capire

**Comment by nikpait** [comment]
Scusa ma per me è la prima volta che sento parlare di VE. 
Ho sempre eseguito l'aggiornamento e non ho mai avuto problemi.
Detto questo mi sai dire qual'è la procedura per aggiornare?

Grazie

**Reply by snakuzzo** [reply]
1. Cortesemente non creare nuovi thread ad ogni risposta, ma rispondi sotto l'ultimo messaggio.
2. Se non sai cosa sia un venv mi aspetto che tu non abbia installato HA Core in venv. Quindi se così fosse, perché stai facendo tutto ciò ?
3. Che tipo di installazione hai?

**Reply by nikpait** [reply]
Ho installato HA su un Raspberry P4
[](url
![HA](https://user-images.githubusercontent.com/55426277/138407538-5dd61065-6515-4376-81d8-906ba1194121.jpg)
)

Stamattina ho provato ad aggornare dalla versione 2021.9.6 alla 2021.10.6 e ho ricevuto dei messaggi di errore uguali a  quello di quest'altro thread "https://github.com/home-assistant/core/issues/57307"

**Reply by snakuzzo** [reply]
Sì però perdonami...tu mi dici "è la prima volta che sento parlare di virtualenv" e poi hai fatto un installazione in virtual env (basta che guardi la voce "Ambiente Virtuale = true" nella schermata che hai postato)...
tutto ciò non ti sembra un po' "assurdo" ?
Vuol dire che hai fatto un'installazione facendo copia-incolla senza nemmeno avere un briciolo di curiosità per capire cosa stavi facendo.

Ad ogni modo...oltre a suggerirti di non usare questo approccio ma di avere curiosità quando fai una cosa con il copia-incolla cercando di capire cosa stai facendo, ti consiglio di leggere [QUI](https://indomus.it/guide/come-installare-e-configurare-home-assistant-su-un-raspberry-pi-gia-in-uso/) (ultimo paragrafo).
In sostanza c'è scritto quello che ti ho detto già questa mattina

**Reply by nikpait** [reply]
Grazie come al solito per l'aiuto, ho risolto seguendo le tue indicazioni.
Detto questo però non è questione di fare l'installzione con copia e incolla. Ho solo eseguito i soliti comandi per aggiornare l' HA come sempre. Ha sempre funzionato senza problemi. Ora se per questo aggiornamento c'è una procedura diversa (Rust Compiler) non lo avevo notato.
Ti ringrazio nuovamente per il tuo supporto

**Reply by snakuzzo** [reply]
Il punto non è rust o "questo aggiornamento".
Il punto è che tu hai fatto un'installazione in venv e non lo sapevi nemmeno ... e questo non va bene

**Comment by LET83** [comment]
Salve è la prima volta scrivo su un forum.
Ho il problema che non riesco ad aggiornare home assistant, quando lancio l-aggiornamento mi dice skipping upgrade.
Non so se qualcuno può aiutarmi

**Reply by Scialla** [reply]
Si però, se vuoi essere aiutato, devi fornire informazioni che noi non abbiamo. Che installazione hai fatto? A che versione sei? che errore ti da il log? Riporta poi il log formattando correttamente.
Che poi sono domande già fatte ad altri in questo stesso topic eh.... rileggi tutte le domande sopra e posta le risposte sotto..
